docs/designers-developers/developers/backward-compatibility/README.md

```diff
@@ -1 +1,57 @@
 # Backward Compatibility
+
+Gutenberg follows the WordPress backwards compatibility principles. This means Backward compatibility for producation public API is guaranteed. In some rare occasions breaking backward compatibility is unavoidable, in that case, the breakage should be as small as possible and the breakage should be documented as clearly as possible to third-party developpers using devnotes.
+
+## What qualifies as a production public API:
+
+Gutenberg code base is compased of two different types of packages: 
+ - **production packages**: these are packages that are shipped in WordPress scripts (example: wp-components, wp-editor...).
+ - **development packages**: these are made of developper tools that can be used by third-party developpers to lint, test, format and build their themes and plugins (example: @wordpress/scrips, @wordpress/env...). Typically, these are consumed as npm dependencies in third-party projects.
+
+ Backward compatibility guarantee only applies to the production packages as updates happen through WordPress upgrades.
+ 
+ Production packages use the `wp` global variable to provide APIs to third-party developpers. These APIs can be JavaScript functions, variables and React components.
+
+### How to guarantee backward compatibility for a JavaScript function
+
+* The name of the function should not change.
+* The order of the arguments of the function should not change.
+* The function's returned value type should not change.
+* Changes to arguments (new arguments, modification of semantics) is possible if we guarantee that all previous calls are still possible.
+
+### How to guarantee backward compatibility for a React Component
+
+* The name of the component should not change.
+* The props of the components should not be removed.
+* Existing prop values should continue to be supported. If a component accept a function as prop, we can update the component to accept a new type for the same prop, but it shouldn't break existing usage.
+* Adding new props is allowed.
+* React Context dependencies can only be added or removed if we ensure the previous context contract is not breaking.
+* Classnames, DOM nodes used inside the tree of the component are not considered part of the public API and can be modified. That said, changes to these should be done with caution as it can affect the styling. When classes need to be renamed, if possible keep the old ones. When possible, document these changes and write a devnote about the changes.
```

Comment:
```suggestion
* Class names and DOM nodes used inside the tree of the component are not considered part of the public API and can be modified. That said, changes to these should be done with caution as it can affect the styling. Keep the old ones if possible. If not, document these changes and write a dev-note about the changes.
```